packages/perseus-core/src/parse-perseus-json/general-purpose-parsers/is-object.ts

```diff
@@ -1,3 +1,18 @@
+// See the tests for an explanation of why we need this.
+const objectConstructorString = Object.prototype.constructor.toString();
+
+const functionToString = Function.prototype.toString;
+
 export function isObject(x: unknown): x is Record<keyof any, unknown> {
-    return x != null && Object.getPrototypeOf(x) === Object.prototype;
+    if (x == null) {
+        return false;
+    }
+    const prototype = Object.getPrototypeOf(x);
+    if (prototype == null) {
+        return true;
+    }
+    return (
+        typeof prototype.constructor === "function" &&
+        functionToString.call(prototype.constructor) === objectConstructorString
+    );
```

Comment:
It's not a lot of code, but it is big-brain code (I rarely think of prototypes on a day-to-day basis anymore). I feel like it would be beneficial to have inline comments to reinforce intent.